Spruce up README docs (#185)
DeterminateSystems/flakehub-push: README.md

    steps:
      - uses: actions/checkout@v4
        with:
          ref: ${{ (inputs.tag != null) && format('refs/tags/{0}', inputs.tag) || '' }}
      - name: Install Nix
        uses: DeterminateSystems/nix-installer-action@main
      - name: Push to FlakeHub
        uses: DeterminateSystems/flakehub-push@main
        with:
          # For the flake's visibility, you can also select "unlisted" if you don't want
          # it to show up in search results and general listings on flakehub.com
          visibility: public
          # Release rolling versions of the form 0.1.* instead of tagged releases
          rolling: true
```

Some other common configuration use cases are described in the sections below, along with a full listing of [all available parameters](#available-parameters).

> #### Set flake visibility to public, private, etc.

Reviewer: ```suggestion
#### Set flake visibility
```

> Whenever you configuration the `flakehub-push` Action, you need to specify the flake's [visibility] using the `visibility` parameter.

Reviewer: ```suggestion
Whenever you configure the `flakehub-push` Action, you need to specify the flake's [visibility] using the `visibility` parameter.
```

| `public`   | The flake is viewable and usable if you know the URL for the flake and it shows up in search results and on the [flake listing][all-flakes].                     |
| `private`  | The flake is viewable and usable only by users who are authenticated and granted access to the flake. Private flakes are available only on [paid plans][signup]. |
| `unlisted` | The flake is viewable and usable only if you know the URL for it. It shows up neither in search results nor on the [flake listing][all-flakes].                  |

#### Rolling releases

For [rolling releases][rolling], as in the example above, set `rolling` to `true`:

```yaml
- uses: DeterminateSystems/flakehub-push@main
  with:
    rolling: true
```

By default, the rolling minor version is 1, meaning that versions are of the form `0.1.[commit count]+rev-[git sha]`.
An example rolling version would be `0.1.1924+rev-ebfe2c639111d7e82972a12711206afaeeda2450`.
You can set a different rolling minor using the `rolling-minor` setting.
This configuration sets the rolling minor to 2:

```yaml
- uses: DeterminateSystems/flakehub-push@main
  with:
    rolling: true
    rolling-minor: 2
```

#### Tagged releases

> Published [tagged releases][tagged] is a little bit tricker because you need to tell `flakehub-push` which tag to use.

Reviewer: ```suggestion
Published [tagged releases][tagged] are a little bit trickier because you need to tell `flakehub-push` which tag to use.
```

Here's an example configuration:

```yaml
on:
  push:
    tags:
      - v?[0-9]+.[0-9]+.[0-9]+*
  workflow_dispatch:
    inputs:
      tag:
        description: The existing tag to publish to FlakeHub
        type: string
        required: true

jobs:
  flakehub-publish:
    runs-on: ubuntu-latest
    permissions:
      id-token: write
      contents: read
    steps:
      - uses: actions/checkout@v4
        with:
          # Checking out only the tag isn't necessary but should speed things up
          ref: ${{ inputs.tag }}
      - uses: DeterminateSystems/nix-installer-action@main
      - uses: DeterminateSystems/flakehub-push@main
        with:
          visibility: private
          tag: ${{ inputs.tag }}
```

#### Store output paths

> [FlakeHub] has a feature called [resolved store paths][store-paths] that, when activated...

Reviewer: ```suggestion
[FlakeHub] has a feature called [resolved store paths][store-paths] that, when activated...
```
Incomplete sentence.

        id: flakehub_push
        with:
          visibility: private
          rolling: true
          include-output-paths: true

      - name: Notify external system
        run: |
          curl -XPOST https://my-recording-system.dev \
            -H "Content-Type: application/json" \
            -H "Bearer: ${{ secrets.RECORDING_SYSTEM_API_KEY }}" \
            -d '{"flake_version":"${{ steps.flakehub_push.outputs.version }}"}'
```

## Developing `flakehub-push`

See the [development docs](./docs/development.md).

[all-flakes]: https://flakehub.com/flakes
[cache]: https://flakehub.com/cache
[curl]: https://curl.se
[flakehub]: https://flakehub.com
[flakes]: https://zero-to-nix.com/concepts/flakes
[gha-outputs]: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/passing-information-between-jobs
[private-flakes]: https://docs.determinate.systems/flakehub/private-flakes
[rolling]: https://docs.determinate.systems/flakehub/concepts/versioning#rolling
[semver]: https://docs.determinate.systems/flakehub/concepts/semver
[signup]: https://flakehub.com/signup
[store-paths]: https://docs.determinate.systems/flakehub/store-paths

> [tagged]: https://docs.determinate.systems/flakehub/concepts/versioning#rolling

Reviewer: ```suggestion
[tagged]: https://docs.determinate.systems/flakehub/concepts/versioning#tagged
```
